skorch/tests/test_callbacks.py

```diff
@@ -527,3 +527,75 @@ def test_print_skipped_if_not_verbose(self, capsys):
 
         stdout = capsys.readouterr()[0]
         assert not stdout
+
+
+@patch('skorch.NeuralNet.save_params', side_effect=lambda x: x)
+class TestCheckpoint:
+    @pytest.yield_fixture
+    def checkpoint_cls(self):
+        from skorch.callbacks import Checkpoint
+        return Checkpoint
+
+    @pytest.fixture
+    def net_cls(self):
+        from skorch.net import NeuralNetRegressor
+        import torch
+
+        class module(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.p = torch.nn.Linear(1,1)
+            def forward(self, x):
+                return self.p(x)
+
+        return partial(
+            NeuralNetRegressor,
+            module=module,
+            max_epochs=10,
+            batch_size=10)
+
+    @pytest.fixture(scope='module')
+    def data(self):
+        # have 10 examples so we can do a nice CV split
+        X = np.zeros((10, 1), dtype='float32')
+        y = np.zeros((10, 1), dtype='float32')
+        return X, y
+
+    def test_none_monitor_saves_always(self, save_params_mock, net_cls, checkpoint_cls, data):
+        net = net_cls(callbacks=[
+            checkpoint_cls(monitor=None),
+        ])
+        net.fit(*data)
+
+        assert save_params_mock.call_count == len(net.history)
+
+    """
+    # TODO: should this fail with KeyError?
+    #       should the user be warned?
```

Comment:
I believe this should raise an error, since making no checkpoint could be very detrimental. The error should be more specific than `KeyError: 'valid_loss_best'`, though.